packages/gatsby/src/redux/reducers/page-data.js

```diff
@@ -0,0 +1,14 @@
+module.exports = (state = new Map(), action) => {
+  switch (action.type) {
+    case `REMOVE_PAGE_DATA`:
+      state.delete(action.payload.id)
+      return state
+
+    case `SET_PAGE_DATA`: {
+      return state.set(action.payload.id, action.payload.resultHash)
+    }
+
+    default:
+      return state
+  }
```

Comment:
I think we need to add `DELETE_CACHE` case here. This should not affect `cachedPageData` we read early on in `build.js`, just make sure that `pageData` substate is generally in sync with other substates.

So something like
https://github.com/gatsbyjs/gatsby/blob/9edffa301adc0e2075c59fd19a4cd62ce94fda16/packages/gatsby/src/redux/reducers/pages.js#L5-L6

This might technically not be needed (as in work might work without it) for now, but it might cause subtle bugs later on when we iterate on code and would expect it to behave similar to other substates

Reply:
Added `DELETE_CACHE`  🙇